hc/api/views.py

```diff
@@ -20,11 +20,14 @@ def ping(request, code):
         check = Check.objects.get(code=code)
     except Check.DoesNotExist:
         return HttpResponseBadRequest()
-
+    
     check.n_pings = F("n_pings") + 1
     check.last_ping = timezone.now()
     if check.status in ("new", "paused"):
         check.status = "up"
+    if check.alert_after and check.last_ping < check.alert_after - check.grace-check.grace:
+        check.status = "often"
+        check.alert_often_pings = True
```

Comment:
The alert_often_pings is what stops the function sendalerts from sending too many alerts when a check is in often mode. without it, the alerts come every second. We were setting it to 'False' after sending one alert.